opm/input/eclipse/EclipseState/Tables/SimpleTable.cpp

```diff
@@ -54,44 +61,50 @@ namespace Opm {
         return result;
     }
 
-    void SimpleTable::addRow( const std::vector<double>& row, const std::string& tableName) {
-        if (row.size() == numColumns()) {
-            for (size_t colIndex  = 0; colIndex < numColumns(); colIndex++) {
-                auto& col = getColumn( colIndex );
-                col.addValue( row[colIndex], tableName );
-            }
-        } else
-            throw std::invalid_argument("Size mismatch");
-    }
+    void SimpleTable::addRow(const std::vector<double>& row,
+                             const std::string& tableName)
+    {
+        const auto ncol = this->numColumns();
 
+        if (row.size() != ncol) {
+            throw std::invalid_argument("Size mismatch");
+        }
 
-    void SimpleTable::addColumns() {
-        for (size_t colIdx = 0; colIdx < m_schema.size(); ++colIdx) {
-            const auto& schemaColumn = m_schema.getColumn( colIdx );
-            TableColumn column(schemaColumn); // Some move trickery here ...
-            m_columns.insert( std::make_pair( schemaColumn.name() , column ));
+        for (auto colIndex = 0*ncol; colIndex < ncol; ++colIndex) {
+            this->getColumn(colIndex).addValue(row[colIndex], tableName);
         }
     }
 
+    void SimpleTable::addColumns()
+    {
+        const auto ncol = this->m_schema.size();
 
-    double SimpleTable::get(const std::string& column  , size_t row) const {
-        const auto& col = getColumn( column );
-        return col[row];
+        for (auto colIdx = 0*ncol; colIdx < ncol; ++colIdx) {
+            const auto& schemaColumn = m_schema.getColumn(colIdx);
+            this->m_columns.insert(std::make_pair(schemaColumn.name(), TableColumn { schemaColumn }));
```

Comment:
for later. add OrderedMap::emplace(). made me twitch..

Reply:
> for later. add `OrderedMap::emplace()`. made me twitch..

Agreed.  I've lamented its absence numerous times.